internal/storage/v2/grpc/tracereader.go

```diff
@@ -73,9 +74,38 @@ func (*TraceReader) FindTraces(
 	panic("not implemented")
 }
 
-func (*TraceReader) FindTraceIDs(
-	context.Context,
-	tracestore.TraceQueryParams,
+func (tr *TraceReader) FindTraceIDs(
+	ctx context.Context,
+	params tracestore.TraceQueryParams,
 ) iter.Seq2[[]tracestore.FoundTraceID, error] {
-	panic("not implemented")
+	return func(yield func([]tracestore.FoundTraceID, error) bool) {
+		query := &storage.TraceQueryParameters{
+			ServiceName:   params.ServiceName,
+			OperationName: params.OperationName,
+			StartTimeMin:  params.StartTimeMin,
+			StartTimeMax:  params.StartTimeMax,
+			DurationMin:   params.DurationMin,
+			DurationMax:   params.DurationMax,
+			SearchDepth:   int32(params.SearchDepth), //nolint: gosec // G115
+		}
+		resp, err := tr.client.FindTraceIDs(ctx, &storage.FindTracesRequest{
+			Query: query,
+		})
+		if err != nil {
+			yield(nil, fmt.Errorf("could not find trace IDs: %w", err))
```

Comment:
misleading, error != "could not find"
```suggestion
			yield(nil, fmt.Errorf("failed to execute FindTraceIDs: %w", err))
```